src/AppInstallerCommonCore/Reboot.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "Public/winget/Reboot.h"
+#include <Windows.h>
+
+namespace AppInstaller::Reboot
+{
+#ifndef AICLI_DISABLE_TEST_HOOKS
+    static bool* s_HasRebootPrivilegeResult_TestHook_Override = nullptr;
+
+    void TestHook_SetHasRebootPrivilegeResult_Override(bool* status)
+    {
+        s_HasRebootPrivilegeResult_TestHook_Override = status;
+    }
+#endif
+
+    bool HasRebootPrivilege()
+    {
+#ifndef AICLI_DISABLE_TEST_HOOKS
+        if (s_HasRebootPrivilegeResult_TestHook_Override)
+        {
+            return *s_HasRebootPrivilegeResult_TestHook_Override;
+        }
+#endif
+
+        wil::unique_handle hToken;
+        TOKEN_PRIVILEGES pTokenPrivileges;
+
+        // Get a token for this process. 
+        if (!OpenProcessToken(GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, &hToken))
+        {
+            return false;
+        }
+
+        // Get the shutdown privilege for this process. 
+        THROW_IF_WIN32_BOOL_FALSE(LookupPrivilegeValue(NULL, SE_SHUTDOWN_NAME, &pTokenPrivileges.Privileges[0].Luid));
```

Comment:
Maybe just return false instead of throw here and below for checking Privilege. Throw will cause the whole process to exit.

Reply:
Changed to just returning false if this fails rather than throwing.